src/main/java/com/powsybl/openloadflow/sa/AbstractSecurityAnalysis.java

```diff
@@ -133,32 +146,102 @@ SecurityAnalysisReport runSync(SecurityAnalysisParameters securityAnalysisParame
+                // run simulation on largest network
+                result = lfNetworks.getLargest().filter(n -> n.getValidity() == LfNetwork.Validity.VALID)
+                        .map(largestNetwork -> runSimulations(largestNetwork, propagatedContingencies, parameters, securityAnalysisParameters, operatorStrategies, actions, limitReductions))
+                        .orElse(createNoResult());
+            }
+        } else {
+            if (!network.getVariantManager().isVariantMultiThreadAccessAllowed()) {
+                throw new PowsyblException("To run a security analysis on multiple threads (and so on variants), you need to set allowVariantMultiThreadAccess to true");
+            }
 
-            stopwatch.stop();
-            LOGGER.info("Security analysis {} in {} ms", Thread.currentThread().isInterrupted() ? "cancelled" : "done",
-                    stopwatch.elapsed(TimeUnit.MILLISECONDS));
+            int partitionSize = contingencies.size() / securityAnalysisParametersExt.getThreadCount();
+            var contingenciesPartitions = Lists.partition(contingencies, partitionSize);
+
+            List<SecurityAnalysisResult> partitionResults = Collections.synchronizedList(new ArrayList<>(Collections.nCopies(contingenciesPartitions.size(), null)));
+            List<LfNetworkList> lfNetworksList = new ArrayList<>();
+
+            Lock networkLock = new ReentrantLock();
+            List<CompletableFuture<Void>> futures = new ArrayList<>();
+            for (int i = 0; i < contingenciesPartitions.size(); i++) {
+                final int partitionNum = i;
+                var contingenciesPartition = contingenciesPartitions.get(i);
+                futures.add(CompletableFuture.runAsync(() -> {
+
+                    var partitionTopoConfig = new LfTopoConfig(topoConfig);
+
+                    //  we have to pay attention with IIDM network multi threading even when allowVariantMultiThreadAccess is set:
+                    //    - variant cloning and removal is not thread safe
+                    //    - we cannot read or write on an exising variant while another thread clone or remove a variant
+                    //    - be aware that even after LF network loading, though LF network we get access to original IIDM
+                    //      variant (for instance to get reactive capability curve), so allowVariantMultiThreadAccess mode
+                    //      is absolutely required
+                    //  so in order to be thread safe, we need to:
+                    //    - lock LF network creation (which create a working variant, see {@code LfNetworkList}
+                    //    - delay {@code LfNetworkList} closing (which remove a working variant) out of worker thread
+                    LfNetworkList lfNetworks;
+                    List<PropagatedContingency> propagatedContingencies;
+                    P parameters;
+                    networkLock.lock();
+                    try {
```

Comment:
I did not get why you need a lock since all threads will set the same variantId. And moreover the variant ID is local to  a thread.
Which function in this section is modifying the Network ?